uniflow/op/model/model_server.py

```diff
@@ -866,6 +898,166 @@ def __call__(self, data: List[str]) -> List[str]:
         return data
 
 
+class SageMakerModelServer(AWSBaseModelServer):
+    """
+    SageMaker Model Server Class.
+
+    The AWS client authenticates by automatically loading credentials as per the methods outlined here:
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html
+
+    If you wish to use a specific credential profile, please provide the profile name from your ~/.aws/credentials file.
+
+    Make sure that the credentials or roles in use have the necessary policies for SageMaker service access.
+    """
+
+    def __init__(
+        self, prompt_template: PromptTemplate, model_config: Dict[str, Any]
+    ) -> None:
+        super().__init__(prompt_template, model_config)
+        self._model_config = SageMakerModelConfig(**self._model_config)
+        self._client = self.session.client(
+            "sagemaker-runtime", region_name=self.aws_region
+        )
+
+    def prepare_input(
+        self, model_type: str, prompt: str, model_kwargs: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Prepare the input for the model based on the model_type.
+
+        Args:
+            model_type (str): The type of the model.
+            prompt (str): The input prompt.
+            model_kwargs (Dict[str, Any]): Additional model arguments.
+
+        Returns:
+            Dict[str, Any]: The prepared input for the model.
+        """
+
+        def prepare_falcon_input(
+            prompt: str, model_kwargs: Dict[str, Any]
+        ) -> Dict[str, Any]:
+            input_body = {
+                "inputs": f"{prompt}",
```

Comment:
qq: why not directly use `"inputs:" prompt`. In this case, there is no difference between `prepare_falcon_input` vs `prepare_default_input`? Also, does all models use the input format like 
`{"inputs" prompt, "parameters": model_kwargs}` or this is model specific?

Reply:
I am still unsure which prompt I should use for the Mistral 7B instruct model/Falcon 7B instruct model, so I have included f"{prompt}" for now, with the intention of updating the prompt later.

Reply:
As for the input format, it relies on the inference code in the SageMaker endpoint. In the notebook, I have specified the input format in the provided inference code, which uses `{"inputs": prompt, "parameters": model_kwargs}` as input.

Reply:
got it. In the HuggingFace implementation running on EC2 [here](https://github.com/CambioML/uniflow/blob/main/uniflow/op/model/model_server.py#L358-L363). we have this specific implementation by adding `# question:   <-- response_start_key is added here !!!` after the `[\INST]` token, could you please help double check regarding how this assume the prompt into the model. I would love to know a bit more details here.